docs/getting_started.md

```diff
@@ -0,0 +1,150 @@
+    --http-address "0.0.0.0:19092" \         # Endpoint for the UI
+    --cluster.peers all.thanos.internal.org  # Discovery of initial cluster peers
+```
+
+The query component is also capable of merging data collected from Prometheus HA pairs. This requires a consistent choice of an external label for Prometheus servers that identifies replicas. Other external labels must be identical. A typical choice is simply the label name "replica" while its value is freely chosable.
+
+Providing the label name to the query component will enable the deduplication.
+
+```
+thanos query \
+    --http-address "0.0.0.0:19092" \
+    --cluster.peers all.thanos.internal.org \
+    --replica-label replica
+```
+
+Go to the configured HTTP address that should now show a UI similar to that of Prometheus itself. If the cluster formed correctly you can now query data across all Prometheus servers within the cluster.
+
+* _[Example Kubernetes manifest](../kube/manifests/thanos-query.yaml)_
+
+## Store Gateway
+
+The sidecar backs up data into your object storage of your choice. But now we also want to query all that data again. The store gateway does just that by implementing the same gRPC data API as the sidecars but backing it with data it can find in your object storage bucket.
+
+Just like sidecars and query nodes, the store gateway joins the gossip cluster and is automatically picked up by running query nodes as yet another data provider.
+
+```
+thanos store \
+    --tsdb.path /var/thanos/store \         # Disk space for local caches
+    --gcs.bucket example-bucket \           # Bucket to fetch data from
+    --cluster.peers all.thanos.internal.org 
+```
+
+The store gateway occupies small amounts of disk space for caching basic information about data in the object storage. This will rarely exceed more than a few gigabytes and is used to improve restart times. It is not useful but not required to preserve it across restarts.
+
+* _[Example Kubernetes manifest](../kube/manifests/thanos-store.yaml)_
+
+## Compactor
+
+A local Prometheus installation periodically compacts older data to improve query efficieny. Since the sidecar backs up data as soon as possible, we need a way to apply the same process to data in the object storage.
+
+The compactor component simple scans the object storage and processes compaction where required. At the same time it is responsible for creating downsampled copies of data to speed up queries.
+
+```
+thanos compact \
+    --data-dir /var/thanos/compact \  # Temporary workspace for data processing
+    --gcs-bucket example-bucket
+```
+
+The compactor is not in the critical path of querying or data backup. It can either be run as a periodic batch job or be left running to always compact data as soon as possible. It is recommended to provide 100-300GB of local disk space for data processing.
+
+_NOTE: The compactor must be run as a **singleton** and must not run when manually modifying data in the bucket._
+
+# All-in-one example
```

Comment:
This doc does not contain `ruler` ... do e intend to add?

Reply:
Yes, we don't really have clear usage path for this, let's add this after our experience with ruler